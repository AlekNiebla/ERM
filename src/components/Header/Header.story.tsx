import * as React from "react";
import { storiesOf } from "@storybook/react";
import Header from "./Header";

storiesOf("Components / Header", module)
  .add("default", () => <Header />)
  .add("with content", () => (
    <Header>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Animi nostrum id
      voluptates nisi ut voluptate voluptatem repellat eos culpa molestiae ipsum
      omnis eaque iusto quisquam, dolorem accusantium earum officiis. Sunt.
    </Header>
  ));
