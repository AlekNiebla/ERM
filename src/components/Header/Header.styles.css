.main {
  background-color: var(--color-gray-lightest);
  display: flex;
  flex-direction: column;
  border-bottom: 1px solid var(--color-gray-lighter);
}
.content {
  padding: var(--step-5);
}
