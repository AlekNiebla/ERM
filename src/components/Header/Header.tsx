import * as React from "react";
// components
import LogoBar from "./../LogoBar/LogoBar";

interface IHeader {
  actionItem?: React.ReactNode;
  children?: React.ReactNode;
}

const Header = (props: IHeader) => (
  <div>
    <LogoBar />
    {props.children}
  </div>
);

export default Header;
