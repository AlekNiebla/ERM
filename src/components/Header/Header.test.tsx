import * as React from "react";
import { shallow } from "enzyme";
// components
import Header from "./Header";
// styles
import * as styles from "./Header.styles.css";

describe("Header component...", () => {
  it("... renders without errors", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(Header)).toBeTruthy();
    expect(wrapper.find(`${styles.main}`)).toBeTruthy();
  });
  it("... matches the snapshot", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
  it("... renders passed children", () => {
    const TestComponent = () => <span>Test</span>;
    const wrapper = shallow(<Header>{TestComponent}</Header>);

    expect(wrapper.find(TestComponent)).toBeTruthy();
  });
});
