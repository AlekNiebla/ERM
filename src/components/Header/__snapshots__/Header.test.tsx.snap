// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component... ...renders without errors 1`] = `ShallowWrapper {}`;
